src/main/java/com/hubspot/smtp/messages/CrlfTerminatingChunkedStream.java

```diff
@@ -0,0 +1,38 @@
+package com.hubspot.smtp.messages;
+
+import java.io.InputStream;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.handler.stream.ChunkedStream;
+
+class CrlfTerminatingChunkedStream extends ChunkedStream {
+  private static final byte CR = '\r';
+  private static final byte LF = '\n';
+  private static final int DEFAULT_CHUNK_SIZE = 8192;
+  private static final byte[] TRAILING_BYTES = { CR, LF };
+
+  CrlfTerminatingChunkedStream(InputStream in) {
+    this(in, DEFAULT_CHUNK_SIZE);
+  }
+
+  CrlfTerminatingChunkedStream(InputStream in, int chunkSize) {
+    super(in, chunkSize);
+  }
+
+  @Override
+  public ByteBuf readChunk(ByteBufAllocator allocator) throws Exception {
+    ByteBuf chunk = super.readChunk(allocator);
+
+    int length = chunk.readableBytes();
+    if (!isEndOfInput()) {
+      return chunk;
+    }
+
+    if (length >= 2 && chunk.getByte(length - 2) == CR && chunk.getByte(length - 1) == LF) {
+      return chunk;
+    }
+
+    return allocator.compositeBuffer(2).addComponents(true, chunk, allocator.buffer(2).writeBytes(TRAILING_BYTES));
```

Comment:
After sending the `MessageContent` we send `EMPTY_LAST_CONTENT` which is an instance of `LastSmtpContent`. Netty's `SmtpRequestEncoder` adds the final `.\r\n` [when it processes that](https://github.com/netty/netty/blob/4.1/codec-smtp/src/main/java/io/netty/handler/codec/smtp/SmtpRequestEncoder.java#L78-L81). 

We could alternatively send `.\r\n` ourselves, but SmtpRequestEncoder also sets a `contentExpected` flag when `DATA` is sent, and this is only unset when `LastSmtpContent` is written.